### ISSUE TITLE
Context type "collectionyourepisodes" throws an error when retrieved using the `current_playback` function

### ISSUE THREAD

**Issue by NDoolan360** [ramsayleung/rspotify#398]
**Describe the bug**
The context type "collectionyourepisodes" throws an error when retrieved using the `current_playback` function.

**To Reproduce**
1. Create any app using rspotify.
2. Using the Spotify desktop app (or other) play an episode from the "Your Episodes" playlist.
3. Have the app created in 1. call the `current_playback` function.

**Expected behavior**
The model is able to handle all types that it could receive for the context type. 

**Log/Output data**
```
ParseJson(
    Error("unknown variant `collectionyourepisodes`, expected one of `artist`, `album`, `track`, `playlist`, `user`, `show`, `episode`, `collection`", line: 19, column: 37),
)
```

**Additional context**
Not sure if this is an issue with rspotify or with the Spotify API.

**Comment by ramsayleung** [comment]
From your error message, it looks like that Spotify has added a new variant `collectionyourepisodes` for the `Type` Enum.

**Comment by NDoolan360** [comment]
This is the state returned when playing an episode from the "Your Episodes" playlist
(truncated some info for both privacy and brevity).

```{
    "actions": {
        "disallows": {
            "resuming": true
        }
    },
    "context": {
        "external_urls": {
            "spotify": "https://open.spotify.com/collection/episodes"
        },
        "href": "https://api.spotify.com/v1/me/episodes",
        "type": "collectionyourepisodes",
        "uri": "spotify:user:<username>:collection:your-episodes"
    },
    "currently_playing_type": "episode",
    "device": {
        ...
    },
    "is_playing": true,
    "item": {
        "is_externally_hosted": false,
        "is_playable": true,
        "type": "episode",
        ...
    },
    "progress_ms": 3515556,
    "repeat_state": "context",
    "shuffle_state": true,
    "timestamp": 1683852851109
}```

**Comment by ramsayleung** [comment]
This problem has been fixed, you could give v0.11.7 a try.
